3-async-await/solution.js

```diff
@@ -22,14 +22,62 @@ Example:
 
 function solution() {
     // YOUR SOLUTION GOES HERE
+    const products = require('./products');
+    const prices = require('./prices');
 
     // You generate your id value here
+    const id = Number((Date.now()+'').slice(-2));
+    const result = {id};
 
-    // You use Promise.all() here
+    (async () => {
+        // You use Promise.all() here
+        console.log('Promise.all:\n')
 
-    // You use Promise.allSettled() here
+        try {
+            [result.product, result.price] = await Promise.all([products(id), prices(id)]);
+            console.log(
+                `  id: ${result.id}\
+                \n  product: ${result.product}\
+                \n  price: ${result.price}`);
+        } catch(reason) {
+            console.log(`   error: ${reason.message}`);
+        }
 
-    // Log the results, or errors, here
+        // You use Promise.allSettled() here
+
+        console.log('\nPromise.allSettled:\n')
+
+        const settled = await Promise.allSettled([products(id), prices(id)]);
+
+        const print = response => 
+            response.status === 'fulfilled' ? response.value : response.reason.message;      
+
+        console.log(
+            `  id: ${id}\
+            \n  product: ${print(settled[0])}\
+            \n  price: ${print(settled[1])}\n`
+        );
+
+        //Promise.race
+
+        console.log('\nPromise.race:\n');
+        try {
+            const value = await Promise.race([products(id), prices(id)]);
+            console.log(`   Is it a product or a price?: ${value}`);
+        } catch (reason) {
+            console.log(`   error: ${reason.message}`);
+        }
+
+        //Promise.any
+
+        console.log('\nPromise.any:\n');
+        try {
+            const value = await Promise.any([products(id), prices(id)]);
+            console.log(`   Is it a product or a price?: ${value}`);
+        } catch (reason) {
+            console.log(`   error: ${reason.message}`);
+        }
+    })();
 }
 
 solution()
```

Comment:
should be async call

Reply:
What is Promise any and race for?

Reply:
Promise any and race are there just to compare visually their output with the Promise.all and Promise.allSettled ones, I did that just to take advantage of all outputs together.